server/controllers/finance/accounts/references.compute.js

```diff
@@ -0,0 +1,184 @@
+          glb.fiscalYear.period_number,
+          glb.fiscalYear.id
+        );
+      });
+      return Q.all(dbPromises);
+    })
+    .then(data => {
+      const accountReferenceValues = data.map(line => line[0]);
+      return accountReferenceValues;
+    });
+}
+
+/**
+ * @method computeSingleAccountReference
+ *
+ * @description
+ * Returns the debit, credit and balance of the account reference given as an array
+ *
+ * @param {string} abbr - the reference of accounts. ex. AA or AX
+ * @param {number} periodId - the period needed
+ * @param {boolean} isAmoDep - the concerned reference is for amortissement, depreciation or provision
+ */
+function computeSingleAccountReference(abbr, isAmoDep = 0, periodId) {
+  const glb = {};
+
+  // get fiscal year information for the given period
+  const queryFiscalYear = `
+    SELECT fy.id, p.number AS period_number FROM fiscal_year fy
+    JOIN period p ON p.fiscal_year_id = fy.id
+    WHERE p.id = ?
+  `;
+
+  const queryAccountReference = `
+    SELECT id, abbr, description, is_amo_dep FROM account_reference
+    WHERE abbr = ? AND is_amo_dep = ?;
+  `;
+
+  return db.one(queryFiscalYear, [periodId])
+    .then(fiscalYear => {
+      glb.fiscalYear = fiscalYear;
+
+      return db.one(queryAccountReference, [abbr, isAmoDep]);
+    })
+    .then(ar => {
+      return getValueForReference(
+        ar.abbr,
+        ar.isAmoDep,
+        ar.description,
+        glb.fiscalYear.period_number,
+        glb.fiscalYear.id
+      );
+    })
+    .then(data => {
+      return data[0];
+    });
+}
+
+/**
+ * @method getValueForReference
+ *
+ * @description
+ * Returns computed value of the reference in a given period and fiscal_year
+ *
+ * @param {number} fiscalYearId
+ * @param {number} periodNumber
+ * @param {string} abbr - the reference of accounts. ex. AA or AX
+ * @param {boolean} isAmoDep - the concerned reference is for amortissement, depreciation or provision
+ */
+function getValueForReference(abbr, isAmoDep = 0, referenceDescription, periodNumber, fiscalYearId) {
+  const queryTotals = `
+  SELECT abbr, is_amo_dep, description,
+    IFNULL(debit, 0) AS debit, IFNULL(credit, 0) AS credit, IFNULL(balance, 0) AS balance FROM (
+      SELECT ? AS abbr, ? AS is_amo_dep, ? AS description,
+        SUM(IFNULL(pt.debit, 0)) AS debit, SUM(IFNULL(pt.credit, 0)) AS credit,
+        SUM(IFNULL(pt.debit - pt.credit, 0)) AS balance
+      FROM period_total pt 
+      JOIN period p ON p.id = pt.period_id
+      WHERE pt.fiscal_year_id = ? AND pt.locked = 0 AND p.number BETWEEN 0 AND ? AND pt.account_id IN (?)
+    )z
+  `;
+
+  return getAccountsForReference(abbr, isAmoDep)
+    .then(accounts => {
+      const accountIds = accounts.map(a => a.account_id);
+      const parameters = [
+        abbr,
+        isAmoDep,
+        referenceDescription,
+        fiscalYearId,
+        periodNumber,
+        accountIds.length ? accountIds : null,
+      ];
+      return db.exec(queryTotals, parameters);
```

Comment:
Your function is called `getValueForReference`, but it returns an array!  It's clear that all of the functions that call this one want a single value, could you either do:
```js
const [value] = await db.exec(queryTotals, parameters);
return value;
```
...or...
```js
return db.exec(queryTotals, parameters).then(values => values[0]);
```

It would probably make this function easier to use.